compiler/rustc_errors/src/lib.rs

```diff
@@ -193,6 +215,19 @@ impl SubstitutionPart {
 }
 
 impl CodeSuggestion {
+    fn get_line<'a>(&self, sf: &'a SourceFile, line_number: usize) -> Option<Cow<'a, str>> {
+        self.transcribe(sf.get_line(line_number))
+    }
+
+    fn transcribe<'a>(&self, text: Option<Cow<'a, str>>) -> Option<Cow<'a, str>> {
+        match self.transcription {
+            Transcription::Copy => text,
+            Transcription::Blank => text.map(|s| {
+                s.chars().map(|c| if c.is_whitespace() { c } else { ' ' }).collect()
```

Comment:
If the goal is to preserve indentation, I think *removing* any non-whitespace characters may be better? For example, in the `async fn` case, you probably want `use` aligned with `async`, not indented to the `fn`.

Reply:
Well, then trailing `... */` comments end up messing up the indentation?

But I freely admit, trailing `... */` comments sound far far more rare than `#[tokio::main]`. :)

Reply:
(I think I'm gradually talking myself into going back and looking at deriving the span from the span of the containing mod/crate again, despite my earlier frustrations with that approach.)